src/main/java/com/cdancy/bitbucket/rest/features/InsightsApi.java

```diff
@@ -0,0 +1,167 @@
+                               @PathParam("commitId") String commitId,
+                               @PathParam("key") String key);
+
+    @Named("insights:delete-annotation")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp11"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}/annotations")
+    @Fallback(BitbucketFallbacks.RequestStatusOnError.class)
+    @ResponseParser(RequestStatusParser.class)
+    @DELETE
+    RequestStatus deleteAnnotation(@PathParam("project") String project,
+                                   @PathParam("repo") String repo,
+                                   @PathParam("commitId") String commitId,
+                                   @PathParam("key") String key,
+                                   @Nullable @QueryParam("externalId") String externalId);
+
+    @Named("insights:create-annotations")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp12"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}/annotations")
+    @Fallback(BitbucketFallbacks.RequestStatusOnError.class)
+    @ResponseParser(RequestStatusParser.class)
+    @POST
+    RequestStatus createAnnotations(@PathParam("project") String project,
+                                    @PathParam("repo") String repo,
+                                    @PathParam("commitId") String commitId,
+                                    @PathParam("key") String key,
+                                    @BinderParam(BindToJsonPayload.class) CreateAnnotations createAnnotations);
+
+    @Named("insights:get-annotations")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp13"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}/annotations")
+    @Fallback(BitbucketFallbacks.AnnotationsResponseOnError.class)
+    @GET
+    AnnotationsResponse getAnnotationsByReport(@PathParam("project") String project,
+                                               @PathParam("repo") String repo,
+                                               @PathParam("commitId") String commitId,
+                                               @PathParam("key") String key);
+
+    @Named("insights:create-annotation")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp13"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}/annotations/{externalId}")
+    @Fallback(BitbucketFallbacks.RequestStatusOnError.class)
+    @ResponseParser(RequestStatusParser.class)
+    @PUT
+    RequestStatus createAnnotation(@PathParam("project") String project,
```

Comment:
:warning: Rather than using a lot of String arguments, consider using a container object for those values. (#QR13:161#)-PMD_UseObjectForClearerAPI